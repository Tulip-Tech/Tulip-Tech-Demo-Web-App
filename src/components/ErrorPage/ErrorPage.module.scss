@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

.errorPage {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 70vh;

  a,
  a:visited,
  a:hover,
  a:active {
    color: inherit;
    font-weight: var(--body-font-weight-bold);
    text-decoration: underline;
    cursor: pointer;
  }
}

.box {
  max-width: 600px;
  padding: 12px;
}

.title {
  margin-bottom: 24px;
  color: variables.$white;
  font-weight: var(--body-font-weight-bold);
  font-size: 34px;
}

.main {
  color: variables.$white;

  > h6 {
    margin-bottom: 16px;
    font-weight: var(--body-font-weight-bold);
    font-size: 20px;
  }
}

@include responsive.mobile-only() {
  .title {
    font-size: 24px;
  }
}

.logo {
  margin-bottom: 24px;
}

.links {
  position: relative;
}

.stack {
  a {
    position: absolute;
    right: 0;
  }
}

.image {
  max-width: 200px;
  max-height: 80px;
  margin-bottom: 24px;
}
