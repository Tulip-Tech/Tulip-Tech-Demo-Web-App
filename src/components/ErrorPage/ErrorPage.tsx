import React, { ReactNode } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './ErrorPage.module.scss';

import { IS_DEMO_MODE, IS_DEVELOPMENT_BUILD, IS_PREVIEW_MODE } from '#src/utils/common';
import DevStackTrace from '#components/DevStackTrace/DevStackTrace';
import { useConfigStore } from '#src/stores/ConfigStore';
import { getPublicUrl } from '#src/utils/domHelpers';

interface Props {
  disableFallbackTranslation?: boolean;
  title?: string | ReactNode;
  message?: string | ReactNode;
  learnMoreLabel?: string;
  children?: React.ReactNode;
  error?: Error;
  helpLink?: string;
}

const ErrorPage = ({ title, message, learnMoreLabel, ...rest }: Props) => {
  const { t } = useTranslation('error');

  return (
    <ErrorPageWithoutTranslation
      title={title || t('generic_error_heading')}
      message={message || t('generic_error_description')}
      learnMoreLabel={learnMoreLabel || t('learn_more')}
      {...rest}
    />
  );
};

export const ErrorPageWithoutTranslation = ({ title, children, message, learnMoreLabel, error, helpLink }: Props) => {
  const logo = useConfigStore((s) => s.config?.assets?.banner);

  return (
    <div className={styles.errorPage}>
      <div className={styles.box}>
        <img className={styles.image} src={getPublicUrl(logo || '/images/logo.png')} alt={'Logo'} />
        <header>
          <h1 className={styles.title}>{title || 'An error occurred'}</h1>
        </header>
        <main className={styles.main}>
          <>
            {children || <p>{message || 'Try refreshing this page or come back later.'}</p>}
            {(IS_DEVELOPMENT_BUILD || IS_DEMO_MODE || IS_PREVIEW_MODE) && helpLink && (
              <p className={styles.links}>
                <a href={helpLink} target={'_blank'} rel={'noreferrer'}>
                  {learnMoreLabel || 'Learn More'}
                </a>
                {(IS_DEVELOPMENT_BUILD || IS_PREVIEW_MODE) && error?.stack && (
                  <span className={styles.stack}>
                    <DevStackTrace error={error} />
                  </span>
                )}
              </p>
            )}
          </>
        </main>
      </div>
    </div>
  );
};

export default ErrorPage;
