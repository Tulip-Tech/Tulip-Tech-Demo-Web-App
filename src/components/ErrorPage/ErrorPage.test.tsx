import React from 'react';
import { render } from '@testing-library/react';

import ErrorPage from './ErrorPage';

describe('<ErrorPage>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ErrorPage title="This is the title">This is the content</ErrorPage>);

    expect(container).toMatchSnapshot();
  });
});
