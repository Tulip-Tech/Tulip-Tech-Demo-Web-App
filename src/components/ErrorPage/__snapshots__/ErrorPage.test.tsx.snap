// Vitest Snapshot v1

exports[`<ErrorPage> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_errorPage_8c5621"
  >
    <div
      class="_box_8c5621"
    >
      <img
        alt="Logo"
        class="_image_8c5621"
        src="/images/logo.png"
      />
      <header>
        <h1
          class="_title_8c5621"
        >
          This is the title
        </h1>
      </header>
      <main
        class="_main_8c5621"
      >
        This is the content
      </main>
    </div>
  </div>
</div>
`;
